src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -154,20 +166,37 @@ private ConstantValue DefaultSyntaxValue
             }
         }
 
+        protected ConstantValue MakeDefaultExpression(DiagnosticBag diagnostics, Binder binder)
+        {
+            var result = MakeDefaultExpression(diagnostics, binder, out var convertedExpression);
+
+            if (binder?.IsSemanticModelBinder != true && (this.ContainingSymbol as MethodSymbol)?.MethodKind != MethodKind.LocalFunction)
```

Comment:
>(this.ContainingSymbol as MethodSymbol)?.MethodKind != MethodKind.LocalFunction [](start = 57, length = 79)

Consider simply checking if ```this.ContainingSymbol.ContainingSymbol.Kind == NamedType```. I think it will be more robust long term.